deltacat/compute/compactor/steps/materialize.py

```diff
@@ -29,7 +29,7 @@
 logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
 
 
-@ray.remote(num_cpus=0.5)
+@ray.remote
 def materialize(
```

Comment:
This isn't a result of this PR, but I noticed that the recent change to eagerly evict tables from materialize memory has introduced a side-effect that, whenever we have more than `max_records_per_output_file` to materialize, we materialize `max_records_per_output_file` to file 1-N (where N will also usually be 1), and then write the remainder to the next file.

This (1) erodes our ability to produce a read-optimized output table and (2) broke the source -> dest primary key index record mapping performed by the actor removed in this PR (since it assumed that materialize will write `max_records_per_output_file` for each file, and then only chop off the remainder once across files assigned to a materialize task). The second doesn't matter if we don't ever go back to mapping records from src -> dest at the end of the `dedupe` step, but the first is still an issue.

We should also capture an issue to fix this in a follow-up PR where, instead of chopping off the remainder over `max_records_per_output_file` every time we met or exceed this record count, we write one file and then continue appending records to the remainder. A remainder with less than `max_records_per_output_file` should only be written when no more records are left for the task to materialize.

Reply:
Agree. The implicit coupling makes the materialize implementation fragile. As long we enforce a contract through some unit tests, any new developer will be reminded of following the contract. For now, have added a TODO.